---
services:
  server:
    image: gitea/gitea:1.24.4
    container_name: gitea
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=gitea
      - GITEA__database__USER=gitea
      - GITEA__database__PASSWD=gitea
    restart: always
    networks:
      - gitea
    volumes:
      - ./gitea:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "3000:3000"
      - "80:80"
      - "222:22"
    depends_on:
      - db

  db:
    image: postgres:14
    restart: always
    environment:
      - POSTGRES_USER=gitea
      - POSTGRES_PASSWORD=gitea
      - POSTGRES_DB=gitea
    networks:
      - gitea
    volumes:
      - ./postgres:/var/lib/postgresql/data

  runner:
    image: gitea/act_runner:0.2.12
    environment:
      # CONFIG_FILE: /config.yaml
      GITEA_INSTANCE_URL: "https://gitea.local.ildoc.it"
      GITEA_RUNNER_REGISTRATION_TOKEN: "WtolDCSHHFGUzbWXJXxekqLdSKcfx8xmbW2jHgue"
      GITEA_RUNNER_NAME: "runner_01"
    depends_on:
      - server
      - db
    restart: always
    networks:
      - gitea
    volumes:
      # - ./runner_config.yaml:/config.yaml
      - ./runner:/data
      - /var/run/docker.sock:/var/run/docker.sock

networks:
  gitea:
    external: false
